src/BleError.js

```diff
@@ -66,13 +66,14 @@ export class BleError extends Error {
   }
 }
 
-export function parseBleError(errorMessage: string): BleError {
-  let bleError: BleError
+export function parseBleError(errorMessage: string, errorMessagesArray: Array<string>): BleError {
+	let bleError: BleError
+	const dictionary = errorMessagesArray ? errorMessagesArray : BleErrorCodeMessage
   try {
     const nativeBleError = JSON.parse(errorMessage)
-    bleError = new BleError(nativeBleError)
+    bleError = new BleError(nativeBleError, dictionary)
```

Comment:
I think that you should use consistent names. Once you are calling it dictionary and once an array. I think that dictionary is more suitable. My Ideas: errorMessageMapping, errorMessageDict, errorMessageMappingDict